app/scripts/platforms/extension.js

```diff
@@ -2,18 +2,36 @@ import extension from 'extensionizer'
 import { createExplorerLink as explorerLink } from 'etherscan-link'
 import { getEnvironmentType, checkForError } from '../lib/util'
 import { ENVIRONMENT_TYPE_BACKGROUND } from '../lib/enums'
+import pify from 'pify'
+
+const CHROME_PIFY_OPTS = {
+  errorFirst: false,
+}
 
 class ExtensionPlatform {
 
+  constructor () {
+
+    if (extension.tabs) {
+      this.createTab = typeof chrome !== 'undefined'
+        ? pify(extension.tabs.create, CHROME_PIFY_OPTS)
+        : extension.tabs.create
```

Comment:
TIL Chrome's WebExtension API implementation uses callbacks while Firefox's uses Promises 😑 

- https://developer.chrome.com/extensions/tabs#method-create
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/create

Reply:
Use callbacks exclusively because Firefox also supports them (though they don't make this clear in their docs).

Probably best to stick with callbacks for now, to match the conventions we use elsewhere. Though eventually I'd like us to switch to a library that abstracts over both, and uses the proper promise API instead for Firefox.

Reply:
Ah, that's helpful to know. Well, I'm happy to remove the `constructor` for that class (at least the conditional), but can I just keep the `pify`? `extension.tabs.create` is only called directly inside that module, and I'd rather not rewrite the function signatures all the way to `openExtensionInBrowserExternal`.

Reply:
Sure, that seems fine to me 👍 

Oh and for reference, here's the polyfill I was hoping to switch to: https://github.com/mozilla/webextension-polyfill
This would kinda replace `extensionizer`. I'm not sure if there's anything `extensionizer` does that'd be worth keeping, or whether we'd want to use the polyfill from extensionizer or something.

Reply:
Oh, that's nice of Mozilla!

I replaced one other call in `extension.js` to `extension.tabs.create` with `this.openWindow`, which uses `pify` internally. Lmk if there's anything you want me to change.